### PR TITLE
Migrate to bevy 0.11

### PR DESCRIPTION
**Pull request by jkb0o**
About
------

This is migration to bevy 0.11 PR. 

A lot of work was done by @PhaestusFox in #63, and this PR is logical continuation of his work. If you want to contribute to bevy-011 version, please make pr to this branch.

You can test this by adding belly to deps:
```bash
# bash
cargo add --git https://github.com/jkb0o/belly.git --branch bevy-0.11 belly
```

```Cargo.toml
# Cargo.toml
[dependencies]
belly = { git = "https://github.com/jkb0o/belly.git", branch = "bevy-0.11", version = "0.2.0" }
```

Tracker
-------

- [x] Migrate most of the code (#63)
- [x] Fix runtime crashes (11858da & 1c53d16)
- [x] Proper organise systems ordering according to new bevy schedules (ed69540)
- [x] Fix text input selection flickering (2029df88)
- [x] Fix relations systems CPU leaking (37d0adb2)
- [x] Fix stylebox rendering (7e17ac4d)
- [x] Adjust examples to the new version (7c258b28e & 59321a4c)
- [ ] Get rid or deprecate `undefined` style properties in favour of removed `Val::Undefined`
- [x] Implement new column & row gap style properties (91e1d58)
- [x] Remove line breaks (337a535)
- [x] Implement grid properties (2e213e8 & 87c6541)
- [ ] Add CHANGELOG.md
- [ ] Update README.md
- [ ] Update props & widgets docs
- [ ] Bump version to 0.3

Known bugs:
-------------

- Some elemennts with `flex-grow: 1` and text inside have invalid size, relates to bevyengine/bevy#9350

### CODE REVIEW

File: crates/belly_core/src/relations/mod.rs [hunk 1]
```diff
@@ -200,20 +184,19 @@ impl BindingSystemsInternal {
...
             TypeId::of::<T>(),
         );
-        if self.watchers.read().unwrap().contains(&watcher) {
-            return;
-        }
-        {
-            let watchers = self.watchers.write().unwrap();
-            if watchers.contains(&watcher) {
-                return;
+        if !self.watchers.read().unwrap().contains(&watcher) {
```
Comment (baitcode) [inline crates/belly_core/src/relations/mod.rs:187]: If two watches would contain two watchers, which watch would watch which watcher

Comment (baitcode) [reply crates/belly_core/src/relations/mod.rs:187]: previous version with early returns was nicer. but yeah, does not work the same.

File: crates/belly_widgets/src/img.rs [hunk 2]
```diff
@@ -4,8 +4,9 @@ use belly_macro::*;
 use bevy::{
     asset::Asset,
     prelude::*,
-    utils::{HashMap, HashSet},
+    // utils::{HashMap, HashSet},
```
Comment (baitcode) [inline crates/belly_widgets/src/img.rs:7]: yack

File: docs/style-properties.md [hunk 3]
```diff
@@ -99,7 +99,7 @@ stylebox-source: "images/stylebox.png"
 <!-- @property-type=$val -->
 Size type representing `bevy::prelude::Val` type. Possible values:
 - `auto` for `Val::Auto`
-- `undefined` for `Val::Undefined`
+- `undefined` for `Val::Px(0.)`
```
Comment (baitcode) [inline docs/style-properties.md:102]: This is kinda weird, right?

File: crates/belly_core/src/input.rs [hunk 4]
```diff
@@ -15,47 +14,30 @@ impl Plugin for ElementsInputPlugin {
         app.add_event::<PointerInput>()
             .add_event::<RequestFocus>()
             .init_resource::<Focused>()
-            .add_system(
-                pointer_input_system
-                    .in_base_set(CoreSet::PreUpdate)
-                    .in_set(Label::Signals)
-                    .after(InputSystem),
+            .add_systems(
```
Comment (baitcode) [inline crates/belly_core/src/input.rs:17]: Can you please add comments on what is happening here? This is the most complex place in whole PR and seems like it's a very important, it defines input processing pipeline, I can't comprehend.

**Review by baitcode — APPROVED** [review]
Disregard my comments on Undefined. I've read PR description. And it's cool.

this is actually 4 PR merged in one
1) update to bevy 0.11
2) step 1 in removal of undefined var
3) style refactoring (min_width and width props moved e.t.c.)
4) update examples
would be easier to maintain and review.

Besides that, great job. Hopefully I'll be able to use this in my game after merge.

File: crates/belly_core/src/eml/build.rs [hunk 5]
```diff
@@ -412,6 +412,7 @@ pub trait Widget {
             ctx.insert(policy);
             ctx.insert(Interaction::default());
         }
+        ctx.insert(Name::new(self.name().as_str()));
```
[the remark this answers is not in the document]
Comment (baitcode) [reply crates/belly_core/src/eml/build.rs:415]: Might be worth adding some prefix so it would be obvious where the name comes from. like "belly_" maybe?

**Review by baitcode — APPROVED** [review]

**Review by baitcode — APPROVED** [review]

File: docs/style-properties.md (hunk 3 above)
Comment (nicoburns) [reply docs/style-properties.md:102]: Yeah, if you need an `undefined` defined value, then it should probably map to `Val::Auto`.

**Pull request merged**